sdk/rust/nym-sdk/src/mixnet/client.rs

```diff
@@ -610,80 +481,57 @@ impl MixnetClient {
     ///
     /// #[tokio::main]
     /// async fn main() {
-    ///     let address = "foobar";
-    ///     let recipient = mixnet::Recipient::try_from_base58_string(address).unwrap();
-    ///     let mut client = mixnet::MixnetClient::connect_new().await.unwrap();
-    ///     let surbs = mixnet::IncludedSurbs::default();
-    ///     client.send_bytes(recipient, "hi".to_owned().into_bytes(), surbs).await;
+    ///     let client = mixnet::MixnetClientBuilder::new()
+    ///         .build::<mixnet::EmptyReplyStorage>()
+    ///         .await
+    ///         .unwrap();
+    ///     let client = client.connect_to_mixnet().await.unwrap();
     /// }
     /// ```
-    pub async fn send_bytes(&self, address: Recipient, message: Vec<u8>, surbs: IncludedSurbs) {
-        let lane = TransmissionLane::General;
-        let input_msg = match surbs {
-            IncludedSurbs::Amount(surbs) => {
-                InputMessage::new_anonymous(address, message, surbs, lane)
-            }
-            IncludedSurbs::ExposeSelfAddress => InputMessage::new_regular(address, message, lane),
-        };
-        self.send(input_msg).await
-    }
+    pub async fn connect_to_mixnet(self) -> Result<MixnetClient>
```

Comment:
Should this function return an error if the socks5 config is set (which strongly indicates a user mistake)?

Reply:
I was thinking that maybe there is a scenario where you would build the client with the socks5 config but later decide not to use it when connecting. But I guess that is just overreaching, making the two flows specifically distinct is probably a better idea. I'll make this throw out an error for that case.